benchmark/utils/timer.hpp

```diff
@@ -213,7 +217,6 @@ class CpuTimer : public Timer {
 
     double toc_impl() override
     {
-        exec_->synchronize();
```

Comment:
CPUs need to keep it though. To explain, there are two ways to time. Let's consider a GPU code (i.e. kernels running).

If you use the CPU based timers (this one), from your POV everything is asynchronous, you submit kernels but don't know anything about when they actually start and when they actually end. To help, what we do is:
+ Synchronize before timing, to ensure nothing is running and we have hope to have a responsive GPU (this one is optional for this phase, but it probably helps in some specific cases).
+ Synchronize before doing the `toc`, so that you know all kernels you launched have completed for real, without this synchronization, because kernel ending and CPUs are non blocking, you would instantly time the end.

When doing GPU timing of a GPU, we are using asynchronous events as well, so here everything happens on the GPU, including the timing, and we use the `eventSynchronize` to ensure get the results back to the CPU. Here we have a pipeline of `eventStart -> kernel -> eventStop -> event2Start -> kernel 2 -> event2Stop`. When doing an `eventSynchronize`, we would still potentially be stalling the pipeline in between kernel and kernel 2 which would give a lesser accuracy than if we did `eventStart -> kernel -> kernel2 -> kerneln -> eventStop`.